constructor/main.py

```diff
@@ -78,7 +79,7 @@ def main_build(dir_path, output_dir='.', platform=cc_platform,
     except ValueError:
         sys.exit("Error: invalid platform string '%s'" % platform)
 
-    construct_path = join(dir_path, 'construct.yaml')
+    construct_path = join(dir_path, construct_yaml_filename)
```

Comment:
```suggestion
    construct_path = join(dir_path, config_file)
```

But! If the input can be relative or absolute paths, this should more carefully join with `os.path.relpath` or similar.

Reply:
This argument is meant to _only_ contain a filename, not a path. It should be combined with the main input directory.

Reply:
But you add a new option which is not validated and doesn't prevent from pass in a path, `--config-file=/some/other/path/construct.yaml` is possible (if I read the code correctly), so you'll have to account for that possibility by constructing the path more defensively. Previously that wasn't needed since the filename was hard-coded.

Reply:
Let's validate then 🚀